lib/gui/modules/drive-scanner.js

```diff
@@ -16,100 +16,150 @@
 
 'use strict';
 
-const Rx = require('rx');
 const os = require('os');
 const _ = require('lodash');
 const EventEmitter = require('events').EventEmitter;
 const drivelist = require('drivelist');
+const debug = require('debug')('etcher:drive-scanner');
 const settings = require('../models/settings');
 
-const DRIVE_SCANNER_INTERVAL_MS = 2000;
-const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 0;
-const emitter = new EventEmitter();
+const DRIVE_SCANNER_DELAY_MS = 2000;
+const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 125;
 
-/* eslint-disable lodash/prefer-lodash-method */
-
-const availableDrives = Rx.Observable.timer(
-  DRIVE_SCANNER_FIRST_SCAN_DELAY_MS,
-  DRIVE_SCANNER_INTERVAL_MS
-)
-
-/* eslint-enable lodash/prefer-lodash-method */
+/**
+ * @summary DriveScanner class
+ * @class
+ * @extends {EventEmitter}
+ * @example
+ * var scanner = new DriveScanner()
+ *   .on('error', (error) => { ... })
+ *   .on('drives', (drives) => { ... })
+ */
+class DriveScanner extends EventEmitter {
```

Comment:
Do we really need a whole class to perform this task? Just worrying that we're adding a lot of complexity (e.g. the number of code lines exploded compared to Rx) to perform a relatively simple task.

Something like this (with a couple extra variables to track things) should implement the same thing:

```js
const scan = (emitter) => {
  return drivelist.scanAsync().then((drives) => {
    emitter.emit('drives', drives);

    if (shouldContinue) {
      return Bluebird.delay(2000).then(scan);
    }
  });
};
```